In [30]:
import apacheLog
import proxyLog, proxyLog2
import importlib
importlib.reload(apacheLog)
importlib.reload(proxyLog)
importlib.reload(proxyLog2)

<module 'proxyLog2' from '/Users/jahseem/SYS-320-SP22/Week2/homework/proxyLog2.py'>

### Apache PHP


In [31]:
apacheLog.apache_events('../../logs/access.log', 'apache', 'php')

404 GET /wordpress/wp-login.php
404 GET /mediawiki/index.php?title=Main_Page


In [32]:
apacheLog.apache_events('../../logs/access.log', 'apache', 'cgi')

404 GET /cgi-bin/test-cgi
404 POST /cgi-bin/test-cgi
404 GET /cgi-bin/htsearch?Exclude=%60/etc/passwd%60


### Proxy Security

In [33]:
proxyLog.proxy_events('../../logs/Proxifier_2k.log', 'proxy', 'attack')

QQ.exe tcpconn3.tencent.com:443  close, 149 bytes  sent,  121  
QQ.exe cgi.qqweb.qq.com:80  close, 477 bytes  sent,  448  
QQ.exe tcpconn.tencent.com:80  open through proxy  proxy.cse.cuhk.edu.hk:5070  HTTPS  
QQExternal.exe proxy.cse.cuhk.edu.hk:5070  close, 1644 bytes  (1.60  KB)  
QQProtectUpd.exe qdun-data.qq.com:443  close, 261 bytes  sent,  70  
QQ.exe tcpconn6.tencent.com:80  close, 0 bytes  sent,  0  
QQ.exe cgi.qqweb.qq.com:80  open through proxy  proxy.cse.cuhk.edu.hk:5070  HTTPS  
QQProtectUpd.exe qd-update.qq.com:8080  error : Could  not  connect  
QQ.exe qqmail.tencent.com:80  close, 336 bytes  sent,  2854  
QQ.exe 2052.flash2-http.qq.com:80  close, 466 bytes  sent,  125682  
QQ.exe showxml.qq.com:80  close, 600 bytes  sent,  1716  
QQ.exe tcpconn3.tencent.com:443  open through proxy  proxy.cse.cuhk.edu.hk:5070  HTTPS  
QQ.exe tcpconn3.tencent.com:80  close, 0 bytes  sent,  0  
QQ.exe tcpconn6.tencent.com:80  open through proxy  proxy.cse.cuhk.edu.hk:5070  HTTPS  
QQ.exe t

In [34]:
proxyLog2.proxy_events('../../logs/Proxifier_2k.log', 'proxy', 'open')

QQ.exe tcpconn.tencent.com:80 open through proxy proxy.cse.cuhk.edu.hk:5070
QQ.exe tcpconn6.tencent.com:443 open through proxy proxy.cse.cuhk.edu.hk:5070
QQ.exe 183.60.49.182:443 open through proxy proxy.cse.cuhk.edu.hk:5070
QQ.exe cgi.qqweb.qq.com:80 open through proxy proxy.cse.cuhk.edu.hk:5070
QQ.exe tcpconn6.tencent.com:80 open through proxy proxy.cse.cuhk.edu.hk:5070
QQPlayer.exe btrace.qq.com:80 open through proxy proxy.cse.cuhk.edu.hk:5070
QQ.exe tcpconn3.tencent.com:443 open through proxy proxy.cse.cuhk.edu.hk:5070
QQProtectUpd.exe qdun-data.qq.com:443 open through proxy proxy.cse.cuhk.edu.hk:5070


### What is your interpretation of the results that you discovered (Did the files match known threat intell information?  Was there activity to and from the same remote host?)

From what I can interpret, it looks like the firewall (proxy in this case) is taking a look at the activity and realied that the QQ.exe was involved in some traffic with the user via HTTP (port 80) and HTTPS (port 443). There was data transferred as well and the proxy made sure to take note of it. There are also signs of large byte numbers from what looks like being taken awayfrom the user accross these ports. It looks as of the attacker was operating on port 5070 from their device. Lastly, this wasn't in the prompt, but I can see that the qq domain has been utilized through other applications such as Wechat and it may be apart of a greater issue I don't fully comprehend yet.

### What went well and what can I improve on ?

Essentially, I enjoyed utilizing these features today because I was able to understand how I woulduse python in more of a real-world - personal application sense. When I usually go over python, it is to make video games or small coding assignments butI feel as if I am doing a lot with a little while also being able to comprehend most of the information that is present. One thing I need to workon tho is my effiency because it took me some time to get back in the groove with what we were taught and how I can apply the class material into a homework setting. That is just me personally, but I am honestly enjoying this class very much and you are doing such a great job! Thanks for the oppurtunity.